src/automaton/modify.rs

```diff
@@ -1,12 +1,12 @@
 use itertools::izip;
-use portgraph::{LinkMut, PortMut, PortView};
+use petgraph::{graph::NodeIndex, Graph};
 
 use crate::{
     predicate::{EdgePredicate, Symbol},
     EdgeProperty, HashSet, PatternID,
 };
 
-use super::{ScopeAutomaton, State, StateID, Transition};
+use super::{view::graph_node_weight, ScopeAutomaton, State, StateID};
 
 impl<PNode: Clone, PEdge: EdgeProperty> ScopeAutomaton<PNode, PEdge> {
     pub(super) fn set_children<I>(
```

Comment:
That's quite the mix of iterables
```rust
preds: impl IntoIterator<Item = I>,
next_states: &[Option<StateID>],
next_scopes: Vec<HashSet<Symbol>>,
```

Could it be this instead?
```rust
preds: impl IntoIterator<IntoIter = EdgePredicate<PNode, PEdge, PEdge::OffsetID>>,
next_states: impl IntoIterator<Item = &Option<StateID>>,
next_scopes: impl IntoIterator<Item = HashSet<Symbol>>,
```

You do not need `ExactSizeIterator` for the output type, and the compiler should be able to use it for `collect` anyways once the method gets monomorphized.

Reply:
Yes, I've actually noticed and fixed this in my current work. I have rewritten a lot of that code, I don't think it's worth fixing here. It will be fixed in the next PR.

Thank you for noticing!